### ISSUE TITLE
CREATE TEMP VIEW and CREATE INDEX ON TEMP TABLE

### ISSUE THREAD

**Issue by derkan** [2ndQuadrant/bdr#121]
Hi,
Does BDR replicates `create temp view` and `create index on temp table`?
Because I see `Database is locked against DDL operations` in other node during this operation.

Version: `postgresql-bdr94-bdr.x86_64` from yum repo on `Centos 7`

Regards

**Comment by ringerc** [comment]
What's your exact version? "SELECT bdr.bdr_version()" please.

Also, please show the exact commands run.
​

**Comment by derkan** [comment]
Version: `0.9.2-2015-07-08-2031bd3`

*One correction*: Index is created on temporary **table**, not temp **view**.

These are sql's running in order:
```sql
CREATE OR REPLACE TEMP VIEW cdrs AS SELECT * FROM ONLY cdrs1 UNION ALL SELECT * FROM ONLY cdrs_zero1;

SET join_collapse_limit = 1;

CREATE TEMPORARY TABLE t1 AS SELECT calls.setup_time, calls.cli, calls.cld,
              cdrs.i_account, cdrs.cost, cdrs.connect_time, cdrs.disconnect_time, cdrs.duration,
              cdrs.cli_in, cdrs.cld_in, cdrs.billed_duration, cdrs.result, cdrs.i_cdr, cdrs.delay, cdrs.remote_ip,
              cdrs.prefix, cdrs.lrn_cld, cdrs.lrn_cld_in, cdrs.area_name
              FROM cdrs JOIN calls USING(i_call) WHERE  calls.setup_time >= '2015-09-08 16:12:34' AND calls.setup_time <= '2015-09-08 23:59:59' AND (cdrs.duration > 0 OR cdrs.result < 0) AND i_account IN (SELECT i_account FROM accounts WHERE accounts.i_customer = '1');

CREATE INDEX t1_i_cdr ON t1(i_cdr);
```

**Comment by derkan** [comment]
I compiled from source(both bdr-plugin and postgresql-bdr). Reinstalled everything. Now version is ` 0.9.2-2015-09-09-9d17926`. But same error messages are in BOTH nodes log file.
Regards,
E.D.

**Comment by ringerc** [comment]
Odd. We check to see if a function affects only temporary objects and ignore it for DDL locking purposes if it does, yet:

```
bdr_testdb=# CREATE TEMPORARY TABLE bob(id integer);
CREATE TABLE
bdr_testdb=# CREATE INDEX test_idx ON bob(id);
ERROR:  database is locked against ddl by another node
HINT:  Node (6195732349873241211,1,16385) in the cluster is already performing DDL
```

Also #124

**Comment by ringerc** [comment]
The issue here is that `statement_affects_only_nonpermanent` doesn't know to check for `T_IndexStmt` or the related statements for drop, etc. Basically just needs to handle more command cases.

**Comment by ringerc** [comment]
I don't expect to get to this very soon due to time requirements for 9.6 development, but if you feel like getting your hands dirty the code in `bdr_commandfilter.c`, function `statement_affects_only_nonpermanent

**Comment by ringerc** [comment]
This is fixed in BDR 1.0.
